common/src/com/thoughtworks/go/config/materials/git/GitMaterial.java

```diff
@@ -96,17 +115,18 @@ public GitMaterial(GitMaterialConfig config) {
 
     @Override
     public MaterialConfig config() {
-        return new GitMaterialConfig(url, branch, submoduleFolder, autoUpdate, filter, folder, name);
+        return new GitMaterialConfig(url, branch, submoduleFolder, autoUpdate, filter, folder, name, shallowClone);
     }
 
     public List<Modification> latestModification(File baseDir, final SubprocessExecutionContext execCtx) {
         ArrayList<Modification> mods = new ArrayList<Modification>();
-        mods.add(getGit(baseDir).latestModification());
+        mods.add(getGit(baseDir, DEFAULT_SHALLOW_CLONE_DEPTH).latestModification());
         return mods;
     }
 
     public List<Modification> modificationsSince(File baseDir, Revision revision, final SubprocessExecutionContext execCtx) {
-        GitCommand gitCommand = getGit(baseDir);
+        GitCommand gitCommand = getGit(baseDir, DEFAULT_SHALLOW_CLONE_DEPTH);
+        unshallowIfNeeded(gitCommand, ProcessOutputStreamConsumer.inMemoryConsumer(), revision, baseDir);
```

Comment:
`modificationsSince` is invoked only by the server. In what cases do think we would need a `unshallowIfNeeded` on the server? `shallowClone` has now become a agent-side property so `unshallowIfNeeded` for agents makes sense. Wondering if I am missing some case.